compiler/luci/pass/src/SubstituteExpandDimsToReshapePass.test.cpp

```diff
@@ -0,0 +1,252 @@
+    _output->shape(shape_out);
+    _output->shape_status(luci::ShapeStatus::VALID);
+    _output->name("output");
+
+    _input->index(_graph_input->index());
+    _output->index(_graph_output->index());
+
+    auto input_shape = std::make_unique<loco::TensorShape>();
+    set(input_shape.get(), shape_in);
+    _graph_input->shape(std::move(input_shape));
+
+    auto output_shape = std::make_unique<loco::TensorShape>();
+    set(output_shape.get(), shape_out);
+    _graph_output->shape(std::move(output_shape));
+  }
+
+protected:
+  void set(loco::TensorShape *shape, const uilist &values)
+  {
+    uint32_t r = 0;
+    shape->rank(values.size());
+    for (auto v : values)
+      shape->dim(r++).set(v);
+  }
+
+public:
+  loco::Graph *g(void) { return &_g; }
+  luci::CircleOutput *output(void) { return _output; }
+
+protected:
+  loco::Graph _g;
+  loco::GraphInput *_graph_input = nullptr;
+  loco::GraphOutput *_graph_output = nullptr;
+  luci::CircleInput *_input = nullptr;
+  luci::CircleOutput *_output = nullptr;
+};
+
+class ExpandDimsWithInputAxisGraph : public PassTestGraph
+{
+public:
+  void init(const uilist shape_in, const uilist shape_out, const uilist shape_axis)
+  {
+
+    PassTestGraph::init(shape_in, shape_out);
+    _graph_axis = _g.inputs()->create();
+
+    _axis = _g.nodes()->create<luci::CircleInput>();
+    _axis->shape(shape_axis);
+    _axis->shape_status(luci::ShapeStatus::VALID);
+    _axis->name("axis");
+
+    _axis->index(_graph_axis->index());
+
+    auto axis_shape = std::make_unique<loco::TensorShape>();
+    set(axis_shape.get(), shape_axis);
+    _graph_axis->shape(std::move(axis_shape));
+
+    _expand_dims = _g.nodes()->create<luci::CircleExpandDims>();
+    _expand_dims->input(_input);
+    _expand_dims->axis(_axis);
+    _expand_dims->name("expand_dims");
+
+    _output->from(_expand_dims);
+  }
+
+protected:
+  loco::GraphInput *_graph_axis = nullptr;
+  luci::CircleInput *_axis = nullptr;
+
+protected:
+  luci::CircleExpandDims *_expand_dims = nullptr;
+};
+
+class ExpandDimsWithConstAxisGraph : public PassTestGraph
+{
+public:
+  void init(const uilist shape_in, const uilist shape_out, const int32_t axis)
+  {
+    PassTestGraph::init(shape_in, shape_out);
+    _axis = luci::create_const_node<int32_t>(g(), loco::DataType::S32, {1}, axis);
+    _axis->name("axis");
+
+    _expand_dims = _g.nodes()->create<luci::CircleExpandDims>();
+    _expand_dims->input(_input);
+    _expand_dims->axis(_axis);
+    _expand_dims->name("expand_dims");
+
+    _output->from(_expand_dims);
+  }
+
+protected:
+  luci::CircleConst *_axis = nullptr;
+
+protected:
+  luci::CircleExpandDims *_expand_dims = nullptr;
+};
+
+class SubstituteExpandDimsToReshapeTest : public ::testing::Test
+{
+public:
+  SubstituteExpandDimsToReshapeTest() = default;
+
+  void run_pass(void)
+  {
+    while (_shapeinf.run(_graph.g()) || _pass.run(_graph.g()))
+      ;
+  }
+
+protected:
+  ExpandDimsWithConstAxisGraph _graph; // *W*ith *C*onst axis
```

Comment:
why use `*W*`, `*C*` ? this is just source code not markdown file.

Reply:
Before I named it with abbreviations. Now it's renamed and I forgot to remove them 😅. Let me remove them :-D